google-cloud-firestore/src/main/java/com/google/cloud/firestore/BasePath.java

```diff
@@ -58,7 +58,10 @@ B append(String path) {
         path != null && !path.isEmpty(), "'path' must be a non-empty String");
     ImmutableList.Builder<String> components = ImmutableList.builder();
     components.addAll(this.getSegments());
-    components.add(splitChildPath(path));
+    components.add(
+        (this instanceof FieldPath && FieldPath.splitOnDots == false)
+            ? new String[] {path}
+            : splitChildPath(path));
     return createPathWithSegments(components.build());
```

Comment:
not a good practice to check the object type (instanceof). You can override `splitChildPath` in the FieldPath

Reply:
@dmitry-fa, ```splitchildpath(path)``` method already present inside FieldPath class [link](https://github.com/suraj-qlogic/java-firestore/blob/b91c57c4b2d3e92478ceaa1a39d467c40e1344dc/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java#L104).Can i write like below?

```
 @Override
  String[] splitChildPath(String path) {
    // By default, split() doesn't return empty leading and trailing segments. This can be enabled
    // by passing "-1" as the  limit.
    return (splitOnDots==true)?path.split("\\.", -1):new String[]{path};
  }
```

Reply:
Yes, you can.

But you don't need  `splitOnDots` field. You can check if the BasePath is empty or not. 

```
   boolean isEmpty = ...;
   return isEmpty ==true ? path.split("\\.", -1):new String[]{path};
```